addons/crm_iap_lead/models/crm_iap_lead_mining_request.py

```diff
@@ -0,0 +1,237 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+
+from odoo import api, fields, models, _
+from odoo.addons.iap import jsonrpc
+
+_logger = logging.getLogger(__name__)
+
+DEFAULT_ENDPOINT = 'https://iap-services.odoo.com'
+MAX_CONTACT = 5
+
+CREDIT_PER_COMPANY = 1
+CREDIT_PER_CONTACT = 1
+
+
+class CRMLeadMiningRequest(models.Model):
+    _name = 'crm.iap.lead.mining.request'
+    _description = 'CRM Lead Mining Request'
+
+    name = fields.Char(string='Request Number', required=True, readonly=True, default=lambda self: _('New'))
+    state = fields.Selection([('draft', 'Draft'), ('pending', 'Pending'), ('done', 'Done')], string='Status', default='draft')
+
+    # Request Data
+    lead_number = fields.Integer(string='Number of Leads', required=True, default=10)
+    search_type = fields.Selection([('companies', 'Companies'), ('people', 'Companies and their Contacts')], string='Target', required=True, default='companies')
+
+    # Lead / Opportunity Data
+    lead_type = fields.Selection([('lead', 'Lead'), ('opportunity', 'Opportunity')], string='Type', required=True, default='lead')
+    team_id = fields.Many2one('crm.team', string='Sales Team', domain="[('use_opportunities', '=', True)]")
+    user_id = fields.Many2one('res.users', string='Salesperson')
+    tag_ids = fields.Many2many('crm.lead.tag', string='Tags')
+    lead_ids = fields.One2many('crm.lead', 'lead_mining_request_id', string='Generated Lead / Opportunity')
+    leads_count = fields.Integer(compute='_compute_leads_count', string='Number of Generated Leads')
+    opportunity_count = fields.Integer(compute='_compute_leads_count', string='Number of Generated Opportunity')
+
+    # Company Criteria Filter
+    filter_on_size = fields.Boolean(string='Filter on Size', default=False)
+    company_size_min = fields.Integer(string='Size', default=0)
+    company_size_max = fields.Integer(default=1000)
+    country_ids = fields.Many2many('res.country', string='Countries', required=True)
+    state_ids = fields.Many2many('res.country.state', string='States')
+    industry_ids = fields.Many2many('crm.iap.lead.industry', string='Industries')
+    technology_ids = fields.Many2many('crm.iap.lead.technology', string='Technologies')
+
+    # Contact Generation Filter
+    contact_number = fields.Integer(string='Number of Contacts', required=True, default=1)
```

Comment:
contact number shouldn't be required on model, as it is not needed at all if you don't want any contact. set an attrs required in view when contact are requested